### PR TITLE
Remove unused GetTimeSeconds

### PR DESCRIPTION
**Pull request by maflcko**
Seems confusing to have this helper when it is possible to get the system time in a type-safe way by simply calling `std::chrono::system_clock::now` (C++11).

This patch replaces `GetTimeSeconds` and removes it:
* in `bitcoin-cli.cpp` by `system_clock`
* in `test/fuzz/fuzz.cpp` by `steady_clock`

### CODE REVIEW

File: src/bitcoin-cli.cpp [hunk 1]
```diff
@@ -433,7 +439,7 @@ class NetinfoRequestHandler : public BaseRequestHandler
         if (conn_type == "addr-fetch") return "addr";
         return "";
     }
-    const int64_t m_time_now{GetTimeSeconds()};
+    const int64_t m_time_now{CountSeconds(Now<CliSeconds>())};
```
Comment (laanwj) [inline src/bitcoin-cli.cpp:442]: As this is used solely to measure intervals, it may be another good use for `steady_clock`.

Comment (laanwj) [reply src/bitcoin-cli.cpp:442]: Oh, never mind, it compares to the server time.

File: src/util/time.h [hunk 2]
```diff
@@ -21,15 +21,20 @@ using SteadyMicroseconds = std::chrono::time_point<std::chrono::steady_clock, st
...
- * This helper is used to convert durations before passing them over an
+ * This helper is used to convert durations/time_points before passing them over an
  * interface that doesn't support std::chrono (e.g. RPC, debug log, or the GUI)
  */
+template <typename Clock>
+constexpr int64_t CountSeconds(std::chrono::time_point<Clock, std::chrono::seconds> t)
+{
+    return t.time_since_epoch().count();
+}
 constexpr int64_t count_seconds(std::chrono::seconds t) { return t.count(); }
```
Comment (laanwj) [inline src/util/time.h:38]: It's confusing to me to have both `count_seconds` and `CountSeconds`.

Comment (MarcoFalke) [reply src/util/time.h:38]: Ah, I forgot that function overloading exists. Thanks, fixed.

**Comment by naumenkogs** [comment]
ACK fab9e8a29c2cdeab6cf1ae7c1fc0e0a3af783b17

**Review by jonatack — COMMENTED** [review]
Post-merge ACK // edit: modulo a few clean-ups addressed in #25155

**Pull request merged**